sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryDilutionDao.java

```diff
@@ -0,0 +1,207 @@
+package uk.ac.bbsrc.tgac.miso.persistence.impl;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.criterion.Criterion;
+import org.hibernate.criterion.Order;
+import org.hibernate.criterion.Projections;
+import org.hibernate.criterion.Restrictions;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import uk.ac.bbsrc.tgac.miso.core.data.AbstractLibrary;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryDilution;
+import uk.ac.bbsrc.tgac.miso.core.data.type.PlatformType;
+import uk.ac.bbsrc.tgac.miso.core.store.LibraryDilutionStore;
+import uk.ac.bbsrc.tgac.miso.sqlstore.util.DbUtils;
+
+public class HibernateLibraryDilutionDao implements LibraryDilutionStore {
+
+  @Autowired
+  private SessionFactory sessionFactory;
+
+  private Session currentSession() {
+    return getSessionFactory().getCurrentSession();
+  }
+
+  private Criterion searchRestrictions(String querystr) {
+    return DbUtils.searchRestrictions(querystr, "name", "identificationBarcode", "library.name", "library.alias", "library.description");
+  }
+
+  @Override
+  public long save(LibraryDilution dilution) throws IOException {
+    long id;
+    if (dilution.getId() == AbstractLibrary.UNSAVED_ID) {
```

Comment:
It's probably the same, but shouldn't be comparing dilution's ID to AbstractLibrary's